quill-sql/src/main/scala/io/getquill/context/sql/idiom/SqlIdiom.scala

```diff
@@ -239,14 +239,15 @@ trait SqlIdiom extends Idiom {
   implicit def propertyTokenizer(implicit valueTokenizer: Tokenizer[Value], identTokenizer: Tokenizer[Ident], strategy: NamingStrategy): Tokenizer[Property] = {
     def unnest(ast: Ast): Ast =
       ast match {
-        case Property(a, b) => unnest(a)
+        case Property(a, _) => unnest(a)
         case a              => a
       }
     Tokenizer[Property] {
-      case Property(ast, "isEmpty")   => stmt"${ast.token} IS NULL"
-      case Property(ast, "nonEmpty")  => stmt"${ast.token} IS NOT NULL"
-      case Property(ast, "isDefined") => stmt"${ast.token} IS NOT NULL"
-      case Property(ast, name)        => stmt"${scopedTokenizer(unnest(ast))}.${strategy.column(name).token}"
+      case Property(ast, "isEmpty")        => stmt"${ast.token} IS NULL"
+      case Property(ast, "nonEmpty")       => stmt"${ast.token} IS NOT NULL"
+      case Property(ast, "isDefined")      => stmt"${ast.token} IS NOT NULL"
+      case Property(Property(ident, _), b) => stmt"${ident.token}.${b.token}"
```

Comment:
It's not clear to me why `unnest` doesn't have the expected effect, since it was meant to be a more generic implementation (for n levels of nesting) of this case statement. Would you know why?